### PR TITLE
Added tests

### PR DESCRIPTION
**Pull request by mprtcz**
Added tests for BoardController and BoardTile

### CODE REVIEW

File: src/test/java/com/spanish_inquisition/battleship/client/board/BoardTileTest.java [hunk 1]
```diff
@@ -37,4 +41,62 @@ public void shouldSetLabels() {
         assertEquals(horizontalLabel, tile.horizontalLabel);
```
Comment (Devas) [inline src/test/java/com/spanish_inquisition/battleship/client/board/BoardTileTest.java]: Use SoftAssert if you have multiple assertions

File: src/test/java/com/spanish_inquisition/battleship/client/board/BoardTileTest.java [hunk 2]
```diff
@@ -37,4 +41,62 @@ public void shouldSetLabels() {
...
         assertEquals(verticalLabel, tile.verticalLabel);
     }
+
+
+
+    @Test
+    public void shouldSetTileStyle_withLabels() {
+        BoardTile boardTile = new BoardTile(1);
```
Comment (Devas) [inline src/test/java/com/spanish_inquisition/battleship/client/board/BoardTileTest.java]: Duplicates, should be extracted to method or make data provider

File: src/test/java/com/spanish_inquisition/battleship/client/board/BoardTileTest.java [hunk 3]
```diff
@@ -37,4 +41,62 @@ public void shouldSetLabels() {
...
+        assertNull(boardTile.horizontalLabel);
+        assertNull(boardTile.verticalLabel);
+        Paint newTestStyle = Styles.TEXT_LIME;
+        String colorStyle = Styles.DEFAULT_TILE_COLOR;
+        boardTile.setTileStyle(colorStyle, newTestStyle);
+    }
+
+    @Test
+    public void shouldReturnBoardIndex() {
+        int boardIndex =5;
```
Comment (Devas) [inline src/test/java/com/spanish_inquisition/battleship/client/board/BoardTileTest.java]: Space, always reformat code

File: src/test/java/com/spanish_inquisition/battleship/client/board/boardcontroller/BoardControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * @author Michal_Partacz
+ */
+public class BoardControllerTest {
+
+    @BeforeSuite
+    public void setupJavaFx() {
```
Comment (Devas) [inline src/test/java/com/spanish_inquisition/battleship/client/board/boardcontroller/BoardControllerTest.java]: :+1:

File: src/test/java/com/spanish_inquisition/battleship/client/board/boardcontroller/BoardControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+            new JFXPanel(); // initializes JavaFX environment
+            latch.countDown();
+        });
+    }
+
+    @Test
+    public void testGetBoardGridPane() throws Exception {
+        GameBoard gameBoard = mock(GameBoard.class);
+        GridPane gridPane = new GridPane();
+        when(gameBoard.getGridPane()).thenReturn(gridPane);
```
Comment (Devas) [inline src/test/java/com/spanish_inquisition/battleship/client/board/boardcontroller/BoardControllerTest.java]: :+1:

File: src/test/java/com/spanish_inquisition/battleship/client/board/boardcontroller/BoardControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    public void testGetGameBoard() throws Exception {
+        GameBoard gameBoard = mock(GameBoard.class);
+        BoardController boardController = new PlayerBoardController(gameBoard, new Game());
+        assertEquals(gameBoard, boardController.getGameBoard());
+    }
+
+    @Test
+    public void testGetBoardsIndexTiles() throws Exception {
+        GameBoard gameBoard = mock(GameBoard.class);
+        Map<Integer, BoardTile> expectedMap = new HashMap<>();
```
Comment (Devas) [inline src/test/java/com/spanish_inquisition/battleship/client/board/boardcontroller/BoardControllerTest.java]: Make data provider

File: src/test/java/com/spanish_inquisition/battleship/client/board/boardcontroller/BoardControllerTest.java [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+        BoardController boardController = new PlayerBoardController(gameBoard, game);
+        boardController.setGame(expectedGame);
+        assertEquals(expectedGame, boardController.getGame());
+    }
+
+    @Test
+    public void testColorBoardTile() throws Exception {
+        GridPane gridPane = mock(GridPane.class);
+        GameBoard gameBoard = new GameBoard(gridPane);
+        BoardTile boardTile1 = new BoardTile(1);
```
Comment (Devas) [inline src/test/java/com/spanish_inquisition/battleship/client/board/boardcontroller/BoardControllerTest.java]: Bad naming, 1,2 doesn't mean anything - maybe currentTile, opponentTile?

File: src/main/java/com/spanish_inquisition/battleship/server/BattleshipServer.java [hunk 8]
```diff
@@ -35,7 +35,8 @@ ServerSocket createServerSocket(int portNumber) {
         ServerSocket serverSocket = null;
         try {
             serverSocket = new ServerSocket(portNumber);
-            logger.log(DEFAULT_LEVEL, "IP ADDRESS OF THE SERVER: " +InetAddress.getLocalHost().toString().split("/")[1]);
+            logger.log(DEFAULT_LEVEL, "IP ADDRESS OF THE SERVER: " +
+                    InetAddress.getLocalHost().toString().split("/")[1]);
```
Comment (piotr921) [inline src/main/java/com/spanish_inquisition/battleship/server/BattleshipServer.java]: '+' should be at start of new line

File: src/main/java/com/spanish_inquisition/battleship/server/ServerMain.java [hunk 9]
```diff
@@ -2,6 +2,7 @@
 
 public class ServerMain {
     public static void main(String[] args) {
+        System.out.println("Server is running...");
```
Comment (piotr921) [inline src/main/java/com/spanish_inquisition/battleship/server/ServerMain.java]: 👍

File: src/test/java/com/spanish_inquisition/battleship/client/board/boardcontroller/BoardControllerTest.java [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+        String style2 = Styles.RESPONSE_DESTROYED;
+        gameBoard.getIndexTiles().put(1, boardTile1);
+        gameBoard.getIndexTiles().put(2, boardTile2);
+        BoardController boardController = new PlayerBoardController(gameBoard, new Game());
+        boardController.colorBoardTile(1, style1);
+        boardController.colorBoardTile(2, style2);
+        assertTrue(boardTile1.getStyle().contains(style1));
+        assertTrue(boardTile2.getStyle().contains(style2));
+
+    }
```
Comment (piotr921) [inline src/test/java/com/spanish_inquisition/battleship/client/board/boardcontroller/BoardControllerTest.java]: 👎 nie pozdrawiam

File: src/main/java/com/spanish_inquisition/battleship/server/BattleshipServer.java (hunk 8 above)
Comment (Dzionek95) [inline src/main/java/com/spanish_inquisition/battleship/server/BattleshipServer.java]: +1

File: src/test/java/com/spanish_inquisition/battleship/client/board/boardcontroller/BoardControllerTest.java (hunk 6 above)
Comment (Dzionek95) [inline src/test/java/com/spanish_inquisition/battleship/client/board/boardcontroller/BoardControllerTest.java]: +1!

**Pull request merged**